gst/tensor_transform/tensor_transform.c

```diff
@@ -1190,27 +1193,18 @@ gst_tensor_transform_transform_caps (GstBaseTransform * trans,
   gst_tensor_config_init (&in_config);
   gst_tensor_config_init (&out_config);
 
-  if (direction == GST_PAD_SINK) {
-    if (gst_tensor_transform_read_caps (caps, &in_config)) {
-      gst_tensor_transform_convert_dimension (filter, direction,
-          &in_config.info, &out_config.info);
-    }
-
-    /**
-     * supposed same framerate from input configuration
-     */
-    out_config.rate_n = in_config.rate_n;
-    out_config.rate_d = in_config.rate_d;
+  if (gst_tensor_transform_read_caps (caps, &in_config)) {
+    gst_tensor_transform_convert_dimension (filter, direction,
+        &in_config.info, &out_config.info);
+  }
 
-    result = gst_tensor_caps_from_config (&out_config);
-  } else {
-    if (gst_tensor_transform_read_caps (caps, &out_config)) {
-      gst_tensor_transform_convert_dimension (filter, direction,
-          &out_config.info, &in_config.info);
-    }
+  /**
+   * supposed same framerate from input configuration
+   */
+  out_config.rate_n = in_config.rate_n;
+  out_config.rate_d = in_config.rate_d;
 
-    result = gst_tensor_caps_from_config (&in_config);
-  }
+  result = gst_tensor_caps_from_config (&out_config);
 
   if (filtercap) {
```

Comment:
Let's add a "cap structure size > 0" check here because you are supplying valid filatercap (othercap) with this commit.

Reply:
I will update your suggestion.

Reply:
Now added the condition  (gst_caps_get_size (filtercap) > 0)